packages/components/src/search-control/index.native.js

```diff
@@ -135,9 +143,11 @@ function SearchControl( {
 	}
 
 	function onCancel() {
-		inputRef.current.blur();
-		clearInput();
-		setIsActive( false );
+		onCancelTimer.current = setTimeout( () => {
```

Comment:
@guarani  So this is an odd one. In the dev tools debugger, I could see the `<TextInput />` in the component tree and there is nothing in the code that would hide or remove that component. 
I had a feeling there might be an Animation issue ( somewhat related to the missing items in the `<FlatList />` search results component )
Well I added a timeout to see if queuing the onCancel calls would do anything and it seems to work, even with a 0 millisecond delay 😄 

I could reproduce the original issue on my Pixel 4 and it's no longer an issue with the timeout there.

Reply:
Oh that's cool, I'll give it a try. I figured since the 0 delay worked all we need to do is run this at the end of the event loop. Sounds like `setImmediate` will do more or less that.

Reply:
Well that didn't work :( . Not sure why but replacing the timeout with `setImmediate`  is causing the inserter bottom sheet to close and then crashes the demo app. Not sure it's worth digging into given setTimeout seems to be working.

Thanks for pointing out that reference doc.

Reply:
I see, it does seem like a strange bug. I'm glad the timeout workaround worked though.
TIL about the `setImmediate` – I'm not clear what it is meant for though. It's interesting how there's a `clearImmediate` too. Reading the link there, I guess that's because you could clear the `setImmediate` before the end of the current "JavaScript block". It looks like it's a [non-standard web API](https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate), but it might be more standard in RN.

Reply:
> Well that didn't work :( . Not sure why but replacing the timeout with setImmediate is causing the inserter bottom sheet to close and then crashes the demo app. I don't think it's worth digging into it right now given setTimeout works (even though it's not the cleanest).

Oh, that's weird, I tried `setImmediate` locally and in my case it worked but I didn't test it on the demo project, only in the WordPress one 🤔 , how did you use it?

> It's interesting how there's a clearImmediate too. Reading the link there, I guess that's because you could clear the setImmediate before the end of the current "JavaScript block". It looks like it's a non-standard web API, but it might be more standard in RN.

Yeah exactly. To be honest, I haven't used it too often but for some cases, it could be useful as a replacement of `setTimeout`.

Reply:
> Oh, that's weird, I tried setImmediate locally and in my case it worked but I didn't test it on the demo project, only in the WordPress one 🤔 , how did you use it?

That is odd, I just tried again on the demo app and it worked fine. 

> It looks like it's a non-standard web API, but it might be more standard in RN.

This is first time I've run across this method. In fact it doesn't look like it's part of any modern browser. I don't see it in Chrome or Firefox. 

Which brings up a consideration. It looks like it's well supported in RN but since we are heavily incorporated into a web project and `setTimeout(fn,0)` is a common web pattern I wonder if it's best to stick with the timeout. Thoughts?

Reply:
I agree that it's not a likely common web pattern but I don't have a strong opinion whether we should use it or not, as a side note, I already used it in the reusable block 😄 :

https://github.com/WordPress/gutenberg/blob/e7df86ec8589cd48ca54e41d37433126094c8777/packages/block-library/src/block/edit.native.js#L140